src/engine.rs

```diff
@@ -1126,4 +1151,112 @@ mod tests {
             });
         }
     }
+
+    #[test]
+    fn test_dump_file_or_directory() {
+        let dir = tempfile::Builder::new()
+            .prefix("test_dump_file_or_directory")
+            .tempdir()
+            .unwrap();
+        let mut file_id = FileId {
+            queue: LogQueue::Rewrite,
+            seq: 7,
+        };
+        let builder = Arc::new(DefaultFileBuilder);
+        let entry_data = vec![b'x'; 1024];
+
+        let mut batches = vec![vec![LogBatch::default()]];
+        let mut batch = LogBatch::default();
+        batch
+            .add_entries::<Entry>(7, &generate_entries(1, 11, Some(&entry_data)))
+            .unwrap();
+        batch.add_command(7, Command::Clean);
+        batch.put(7, b"key".to_vec(), b"value".to_vec());
+        batch.delete(7, b"key2".to_vec());
+        batches.push(vec![batch.clone()]);
+        let mut batch2 = LogBatch::default();
+        batch2.put(8, b"key3".to_vec(), b"value".to_vec());
+        batch2
+            .add_entries::<Entry>(8, &generate_entries(5, 15, Some(&entry_data)))
+            .unwrap();
+        batches.push(vec![batch, batch2]);
+
+        for bs in batches.iter_mut() {
+            let file_path = file_id.build_file_path(dir.path());
+            // Write a file.
+            let mut writer =
```

Comment:
You shouldn't use writer here. Refer to other tests in this file, use `Engine::write` instead.

Reply:
@yuqi1129 Please address this comment.

Reply:
@tabokie , I have move the test to file `engine.rs` and you mean i should put it in `mod.rs`

> use Engine::write instead 

OK, I miss it.

Reply:
> you mean i should put it in `mod.rs`

No, the placement is fine.